platform-sdk/swirlds-logging-log4j-appender/src/main/java/com/swirlds/logging/log4j/appender/SwirldsLogAppender.java

```diff
@@ -0,0 +1,198 @@
+ */
+
+package com.swirlds.logging.log4j.appender;
+
+import com.swirlds.logging.api.Level;
+import com.swirlds.logging.api.Marker;
+import com.swirlds.logging.api.extensions.event.LogEventConsumer;
+import com.swirlds.logging.api.extensions.event.LogEventFactory;
+import com.swirlds.logging.api.extensions.provider.LogProvider;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.Serializable;
+import java.util.Objects;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.Property;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+
+/**
+ * SwirldsLogAppender is a custom log appender for the Log4j2 logging framework that integrates
+ * with the swirlds-logging API. It allows Log4j2 log events to be consumed by the swirlds-logging API.
+ * This appender is specifically designed convert Log4j2 logging events to swirlds-logging events.
+ * <p>
+ * It supports dynamic installation of {@link LogEventFactory} and {@link LogEventConsumer} to
+ * facilitate custom log event creation and consumption.
+ * </p>
+ * <p>
+ *     To forward all logging from Log4j2 to the swirlds-logging API, the following configuration should be used:
+ *     <pre>
+ *         {@code
+ *         <Configuration status="WARN">
+ *             <Appenders>
+ *               <SwirldsLoggingAppender name="SwirldsAppender">
+ *               </SwirldsLoggingAppender>
+ *             </Appenders>
+ *             <Loggers>
+ *               <Root level="all">
+ *                 <AppenderRef ref="SwirldsAppender"/>
+ *               </Root>
+ *             </Loggers>
+ *         </Configuration>
+ *        }
+ *     </pre>
+ */
+@Plugin(name = SwirldsLogAppender.APPENDER_NAME, category = "Core", elementType = "appender", printObject = true)
+public class SwirldsLogAppender extends AbstractAppender implements LogProvider {
+    /**
+     * The name of the appender for Log4j2 configuration.
+     */
+    public static final String APPENDER_NAME = "SwirldsLoggingAppender";
+
+    /**
+     * The log event factory to create log events.
+     * This is set by the swirlds-logging API.
+     */
+    private static LogEventFactory logEventFactory;
+    /**
+     * The log event consumer to consume log events.
+     * This is set by the swirlds-logging API.
+     */
+    private static LogEventConsumer logEventConsumer;
+
+    /**
+     * Constructs a new SwirldsLogAppender instance with default values.
+     * This constructor is used by the swirlds-logging API.
+     */
+    public SwirldsLogAppender() {
```

Comment:
I dislike the pattern of having 2 constructors and 2 instances.
both (log4J + our logging) does not need to create a new one. We can simply have a singleton here.